src/modules/bootloader/main.py

```diff
@@ -314,6 +349,37 @@ def get_grub_efi_parameters():
     return None
 
 
+def run_grub_mkconfig(output_file):
+    """
+    Runs grub-mkconfig in the target environment
+
+    :param output_file: A string containing the path to the generating grub config file
+    :return:
+    """
+
+    # get the partition from global storage
+    partitions = libcalamares.globalstorage.value("partitions")
+    if not partitions:
+        libcalamares.utils.error("Failed to run grub-mkconfig, no partitions defined in global storage")
+        return
+
+    # check for zfs
+    is_zfs = False
+    for partition in partitions:
+        if partition["mountPoint"] == "/" and partition["fs"] == "zfs":
+            is_zfs = True
```

Comment:
There's possibly a more-Pythonic `is_zfs = any([partition["mountPoint"] == "/" and partition["fs"] == "zfs" for partition in partitions])` .. and since this particular check for root-filesystem-is-zfs shows up several times, maybe it should be a free function (e.g. `is_zfs_root(partition)` .. and that alongside a `is_btrfs_root(partition)` for symmetry). Your call, it just strikes me as a duplicated condition that can have a meaningful name attached.